src/lib/operators/jit_operator/jit_types.hpp

```diff
@@ -83,14 +84,45 @@ class JitVariantVector {
   bool is_null(const size_t index) { return _is_null[index]; }
   void set_is_null(const size_t index, const bool is_null) { _is_null[index] = is_null; }
 
+  // Adds an element to the internal vector for the specified data type.
+  // The initial value can be set to Zero, MaxValue, or MinValue in a data type independent way.
+  // The implementation will then construct the concrete initial value for the correct data type using
+  // std::numeric_limits.
+  template <typename T>
+  size_t grow_by_one(const InitialValue initial_value);
+
+  // Adds an element to the internal _is_null vector.
+  size_t grow_is_null_by_one(const bool initial_value) {
```

Comment:
This looks like it is only used internally. Shouldn't it be private so that the value and null vectors don't end up with different lengths? Also, why is this (and other methods) implemented in the hpp file?

Reply:
It was used by the `jit_grow_by_one` stand-alone function. I moved it to the `JitVariantVector` now so the vectors stay in sync.
Also moved all method definitions to the cpp file.